allennlp/data/token_indexers/dep_label_indexer.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from typing import Dict, List
+
+from overrides import overrides
+
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.common import Params
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers.token_indexer import TokenIndexer
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@TokenIndexer.register("dependency_label")
+class DepLabelIndexer(TokenIndexer[int]):
+    """
+    This :class:`TokenIndexer` represents tokens by their syntactic dependency label, as determined
+    by the ``dep_`` field on ``Token``.
+
+    Parameters
+    ----------
+    namespace : ``str``, optional (default=``dep_labels``)
+        We will use this namespace in the :class:`Vocabulary` to map strings to indices.
+    """
+    # pylint: disable=no-self-use
+    def __init__(self, namespace: str = 'dep_labels') -> None:
+        self.namespace = namespace
+
+    @overrides
+    def count_vocab_items(self, token: Token, counter: Dict[str, Dict[str, int]]):
+        if not token.dep_:
+            logger.warning("Token had no dependency label: %s", token.text)
+        counter[self.namespace][token.dep_] += 1
```

Comment:
This will break the vocabulary creation if we count `None` elements, because when we add the token to the namespace, we typecheck that it is a string. I think there should be an `else` case here.

Reply:
Also if this can happen, we should throw an error in case the user tries to use a non-padded namespace. Hmmm. Although that would make it impossible to use for prediction. Maybe not then.

Reply:
Yeah, good catch on the `None`.  For the padded namespace issue, this will actually be non-padded by default ("pos_tags" ends in "tags").  This is probably ok, because presumably you have a small POS tag set (or dependency label set), and you should see all of them at training time.